### PR TITLE
chore: remove deprecated success colors

### LINKED ISSUE #24967: chore: remove all instances of deprecated success colors
### What is this about?

In preparation for upgrading the `@metamask/design-tokens` library to version 4, we need to remove all instances of the deprecated `--color-success-alternative` and `--color-success-disabled`. This color has been removed in v4 and its presence will cause breakages in the application.

[Design tokens migration v3 => v4 documentation](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400)

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Search the codebase for all instances of `--color-success-alternative` and `--color-success-disabled`.
- Replace `--color-success-alternative` and `--color-success-disabled` with an appropriate alternative from the updated design tokens.
- Ensure all components using this color are updated accordingly and tested for visual consistency.
- Update any documentation or style guides referencing `--color-success-alternative` and `--color-success-disabled`.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- All instances of `--color-success-alternative` and `--color-success-disabled` are removed from the codebase.
- The application functions correctly without any references to `--color-success-alternative` and `--color-success-disabled`.
- Visual appearance of components previously using `--color-success-alternative` and `--color-success-disabled` is consistent with design specifications.
- No breaking changes introduced in the application due to this update.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify) 

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
This pull request removes the deprecated success colors (`success-alternative` and `success-disabled`) in preparation for upgrading to [design token v4](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25122?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24967

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Go to all stories that did use the success alternative and success disabled
3. See color updates are working as expected

## **Screenshots/Recordings**

Before/After screenshots in code comments for easier review

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .storybook/3.COLORS.stories.mdx [hunk 1]
```diff
@@ -111,7 +111,6 @@ var(--color-success-default)
 var(--color-success-alternative) [DEPRECATED]
 var(--color-success-muted)
 var(--color-success-inverse)
-var(--color-success-disabled) [DEPRECATED]
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:114]: No `success-disabled` left in the codebase

Screenshot 2024-06-06 at 1 59 28 PM

File: ui/components/component-library/header-base/header-base.stories.tsx [hunk 2]
```diff
@@ -303,7 +303,7 @@ export const UseCaseDemos = (args) => (
         marginBottom={4}
         startAccessory={
           <Button
-            backgroundColor={BackgroundColor.successAlternative}
+            backgroundColor={BackgroundColor.successDefault}
             size={ButtonSize.Sm}
           >
             Unlock
```
Comment (georgewrmarshall) [inline ui/components/component-library/header-base/header-base.stories.tsx:309]: Screenshot 2024-06-06 at 2 07 54 PM
Screenshot 2024-06-06 at 2 08 24 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 3]
```diff
@@ -139,7 +139,7 @@ export const ScrollingStory = (args) => (
       >
         Connect
       </Header>
-      <Content background={BackgroundColor.successAlternative}>
+      <Content background={BackgroundColor.successDefault}>
         {scrollingContent}
       </Content>
       <Footer backgroundColor={BackgroundColor.warningAlternative}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:145]: Screenshot 2024-06-06 at 2 09 41 PM
Screenshot 2024-06-06 at 2 10 12 PM

File: ui/helpers/constants/design-system.ts [hunk 4]
```diff
@@ -32,10 +32,8 @@ export enum Color {
   warningInverse = 'warning-inverse',
   warningDisabled = 'warning-disabled',
   successDefault = 'success-default',
-  successAlternative = 'success-alternative',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:35]: Screenshot 2024-06-06 at 2 11 32 PM

File: ui/helpers/constants/design-system.ts [hunk 5]
```diff
@@ -32,10 +32,8 @@ export enum Color {
...
   successMuted = 'success-muted',
   successInverse = 'success-inverse',
-  successDisabled = 'success-disabled',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:38]: Screenshot 2024-06-06 at 2 11 43 PM

File: ui/pages/onboarding-flow/create-password/index.scss [hunk 6]
```diff
@@ -10,7 +10,7 @@
   }
 
   &__strong {
-    color: var(--color-success-alternative);
+    color: var(--color-success-default);
   }
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/create-password/index.scss:14]: Screenshot 2024-06-06 at 2 14 04 PM
Screenshot 2024-06-06 at 2 14 28 PM
Screenshot 2024-06-06 at 2 15 10 PM
Screenshot 2024-06-06 at 2 14 54 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 7]
```diff
@@ -55,7 +55,7 @@ export const HeaderStory = (args) => (
       >
         Connect
       </Header>
-      <Content backgroundColor={BackgroundColor.successAlternative}>
+      <Content backgroundColor={BackgroundColor.successDefault}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:58]: Screenshot 2024-06-06 at 2 22 12 PM
Screenshot 2024-06-06 at 2 20 43 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 8]
```diff
@@ -76,7 +76,7 @@ export const FullscreenStory = (args) => (
       >
         Connect
       </Header>
-      <Content background={BackgroundColor.successAlternative}>
+      <Content backgroundColor={BackgroundColor.successDefault}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:79]: Screenshot 2024-06-06 at 2 22 31 PM
Screenshot 2024-06-06 at 2 20 59 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 9]
```diff
@@ -109,7 +109,7 @@ export const HeaderFooterStory = (args) => (
       >
         Connect
       </Header>
-      <Content backgroundColor={BackgroundColor.successAlternative}>
+      <Content backgroundColor={BackgroundColor.successDefault}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:112]: Screenshot 2024-06-06 at 2 22 48 PM
Screenshot 2024-06-06 at 2 21 26 PM

File: ui/components/multichain/pages/page/page.stories.tsx [hunk 10]
```diff
@@ -139,7 +139,7 @@ export const ScrollingStory = (args) => (
       >
         Connect
       </Header>
-      <Content background={BackgroundColor.successAlternative}>
+      <Content backgroundColor={BackgroundColor.successDefault}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/pages/page/page.stories.tsx:142]: Screenshot 2024-06-06 at 2 23 05 PM
Screenshot 2024-06-06 at 2 21 45 PM

**Review by brianacnguyen — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**